sympy/stats/stochastic_process_types.py

```diff
@@ -625,9 +680,20 @@ class DiscreteMarkovChain(DiscreteTimeStochasticProcess, MarkovProcess):
 
     def __new__(cls, sym, state_space=S.Reals, trans_probs=None):
         sym = _symbol_converter(sym)
-        state_space = _set_converter(state_space)
-        if trans_probs != None:
+
+        # maybe make trans_probs a MatrixSymbol if it is None
+        if trans_probs is not None:
             trans_probs = _matrix_checks(trans_probs)
+
+        if (state_space is S.Reals) and (trans_probs is not None):
+            # handle symbolic-sized matrix
+            n = trans_probs.shape[0]
+            if isinstance(n, Symbol):
+                state_space = Range(0, n)
```

Comment:
It might not be a symbol but be in an expression involving symbols e.g. `n + 1`. A better test would be
```
if not isinstance(n, Integer):
```